File: compiler/DOC/BYTECODE/MS-Academic-Days-2004-04-20/examples/Calculator.cs
/*
  csc -nologo nested-add.cs
*/

public class Calculator: System.Object {
  int R;
  public class add: System.Object { 
    Calculator origin;
    public int V;
    public add(Calculator outer) { origin = outer; }
    public void Enter(int a) { V = a; }
    public void Do() { origin.R = origin.R + V; }
    public int Exit() { return origin.R; }
  }
  public void set(int V) { R = V; }
  public int call_add(int V){ 
    add A = new add(this);
    A.Enter(V);
    A.Do();
    return A.Exit();
  }
  public static void Main(System.String[] args){
      Calculator C = new Calculator();
      int x;

      System.Console.WriteLine("Using add as a method");
      C.set(12);
      x = C.call_add(5);
      System.Console.WriteLine(x);

      System.Console.WriteLine("Using add as a class");
      Calculator.add A = new Calculator.add(C);
      C.set(17);
      A.Enter(13); 
      A.Do();
      x = A.Exit();
      System.Console.WriteLine(x);
      A.V = 10;
      A.Do();
      x = A.Exit();
      System.Console.WriteLine(x);
    }
}

File: compiler/DOC/BYTECODE/MS-Academic-Days-2004-04-20/examples/Calculator.java
/*
  javac Calculator.java
  java Calculator
*/

public class Calculator extends java.lang.Object {
  public int R;
  public class add extends java.lang.Object { 
    public int V;
    public void Enter(int a) { V = a; }
    public void Do() { R = R + V; }
    public int Exit() { return R; }
  }
  public void set(int V) { R = V; }
  public int add(int V){ 
    add A = new add();
    A.Enter(V);
    A.Do();
    return A.Exit();
  }
  public static void main(java.lang.String[] args){
      Calculator C = new Calculator();
      int x;

      System.out.println("Using add as a method");
      C.set(12);
      x = C.add(5);
      System.out.println(x);

      System.out.println("Using add as a class");
      Calculator.add A = C.new add();
      C.set(17);
      A.Enter(13); 
      A.Do();
      x = A.Exit();
      System.out.println(x);
      A.V = 10;
      A.Do();
      x = A.Exit();
      System.out.println(x);
    }
}
